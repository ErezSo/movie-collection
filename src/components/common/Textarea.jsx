import React from "react";
import PropTypes from "prop-types";

const styles = {
  textarea: {
    width: "60%" 
  }
}

const TextInput = ({ name, label, onChange, placeholder, value }) =>
  <div className="form-group">
    <label htmlFor={name}>
      {label}
    </label>
    <div className="field">
      <textarea
        type="text"
        name={name}
        className="form-control"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        style={styles.textarea}
        rows="10"
      />
    </div>
  </div>;

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string
};

export default TextInput;
