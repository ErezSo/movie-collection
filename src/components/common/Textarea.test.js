import React from "react";
import { shallow } from "enzyme";
import Textarea from "./Textarea";

describe("Textarea", () => {
  const props = {
    name: "name",
    label: "label",
    onChange: jest.fn(),
    placeholder: "placeholder",
    value: "value"
  };
  it("should render", () => {
    const wrapper = shallow(<Textarea {...props} />);
    expect(wrapper).toHaveLength(1);
  });

  it("renders properly", () => {
    const tree = shallow(<Textarea {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
