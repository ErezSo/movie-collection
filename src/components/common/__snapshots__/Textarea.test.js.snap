// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea renders properly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="form-group"
>
    <label
        htmlFor="name"
    >
        label
    </label>
    <div
        className="field"
    >
        <textarea
            className="form-control"
            name="name"
            onChange={[Function]}
            placeholder="placeholder"
            rows="10"
            style={
                Object {
                    "width": "60%",
                  }
            }
            type="text"
            value="value"
        />
    </div>
</div>,
  "nodes": Array [
    <div
      className="form-group"
>
      <label
            htmlFor="name"
      >
            label
      </label>
      <div
            className="field"
      >
            <textarea
                  className="form-control"
                  name="name"
                  onChange={[Function]}
                  placeholder="placeholder"
                  rows="10"
                  style={
                        Object {
                              "width": "60%",
                            }
                  }
                  type="text"
                  value="value"
            />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <TextInput
        label="label"
        name="name"
        onChange={[Function]}
        placeholder="placeholder"
        value="value"
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "label": "label",
          "name": "name",
          "onChange": [Function],
          "placeholder": "placeholder",
          "value": "value",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="form-group"
>
          <label
                    htmlFor="name"
          >
                    label
          </label>
          <div
                    className="field"
          >
                    <textarea
                              className="form-control"
                              name="name"
                              onChange={[Function]}
                              placeholder="placeholder"
                              rows="10"
                              style={
                                        Object {
                                                  "width": "60%",
                                                }
                              }
                              type="text"
                              value="value"
                    />
          </div>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="form-group"
>
          <label
                    htmlFor="name"
          >
                    label
          </label>
          <div
                    className="field"
          >
                    <textarea
                              className="form-control"
                              name="name"
                              onChange={[Function]}
                              placeholder="placeholder"
                              rows="10"
                              style={
                                        Object {
                                                  "width": "60%",
                                                }
                              }
                              type="text"
                              value="value"
                    />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TextInput
    label="label"
    name="name"
    onChange={[Function]}
    placeholder="placeholder"
    value="value"
/>,
}
`;
